katz/src/main/kotlin/katz/Option.kt

```diff
@@ -134,3 +136,23 @@ sealed class Option<out A> {
  * @param default  the default expression.
  */
 fun <B> Option<B>.getOrElse(default: () -> B): B = fold({ default() }, { it })
+
+/**
+ * Companion interface for [Option] type
+ * outside due to https://youtrack.jetbrains.com/issue/KT-10532
+ */
+private interface OptionInstances {
+    fun <A> HK<Option.F, A>.ev(): Option<A> = this as Option<A>
+
+    fun monad() = object : Monad<Option.F> {
+        override fun <A, B> map(fa: HK<Option.F, A>, f: (A) -> B): HK<Option.F, B> =
```

Comment:
Question:

- `Applicative` implements `map` method and delegates its behaviour to  `ap` and `pure` methods.
- `Monad` implement `ap` method and delegates his behaviour to `flatMap` method.
- These kinds of interfaces implements `pure` and `flatMap` behaviours.

In this case, is override `map` method needed?